flink-yarn/src/test/java/org/apache/flink/yarn/TestingYarnAMRMClientAsync.java

```diff
@@ -52,6 +53,9 @@
 	private volatile Consumer<Integer> setHeartbeatIntervalConsumer;
 	private volatile TriFunction<String, Integer, String, RegisterApplicationMasterResponse> registerApplicationMasterFunction;
 	private volatile TriConsumer<FinalApplicationStatus, String, String> unregisterApplicationMasterConsumer;
+	private final CompletableFuture<Void> clientInitFuture = new CompletableFuture<>();
+	private final CompletableFuture<Void> clientStartFuture = new CompletableFuture<>();
+	private final CompletableFuture<Void> clientStopFuture = new CompletableFuture<>();
```

Comment:
Let's replace the futures with consumers and runnables, leaving defining the behaviors to the callers.

Same for `TestingYarnNMClientAsync`.